File: src-tauri/src/commands/tool_config.rs
//! 配置文件管理命令（Claude Code / Codex 渠道的 抽取/存储/应用/覆盖）。

use serde_json::Value;
use tauri::AppHandle;

use crate::error::AppResult;
use crate::models::tool_config::{
    ChannelData, ChannelMeta, ClaudeOperationFields, CodexOperationFields, ExtractResult,
    SaveChannelRequest,
};
use crate::modules::tool_config::{self as tc, Tool};

#[tauri::command]
pub fn list_profile_channels(app: AppHandle, app_type: String) -> AppResult<Vec<ChannelMeta>> {
    tc::list_channels(&app, Tool::from_str(&app_type)?)
}

#[tauri::command]
pub fn get_profile_channel(
    app: AppHandle,
    app_type: String,
    id: String,
) -> AppResult<ChannelData> {
    tc::get_channel(&app, Tool::from_str(&app_type)?, &id)
}

#[tauri::command]
pub fn save_profile_channel(
    app: AppHandle,
    app_type: String,
    req: SaveChannelRequest,
) -> AppResult<ChannelMeta> {
    tc::save_channel(&app, Tool::from_str(&app_type)?, req)
}

#[tauri::command]
pub fn delete_profile_channel(app: AppHandle, app_type: String, id: String) -> AppResult<()> {
    tc::delete_channel(&app, Tool::from_str(&app_type)?, &id)
}

#[tauri::command]
pub fn extract_source_record(app: AppHandle, app_type: String) -> AppResult<ExtractResult> {
    tc::extract_record(&app, Tool::from_str(&app_type)?)
}

#[tauri::command]
pub fn apply_profile_config(
    app: AppHandle,
    app_type: String,
    snapshot: Value,
) -> AppResult<()> {
    tc::apply_config(&app, Tool::from_str(&app_type)?, snapshot)
}

#[tauri::command]
pub fn preview_claude_settings(base: Value, fields: ClaudeOperationFields) -> AppResult<Value> {
    Ok(tc::claude::merge_operation_fields(&base, &fields))
}

#[tauri::command]
pub fn parse_claude_fields(snapshot: Value) -> AppResult<ClaudeOperationFields> {
    Ok(tc::claude::parse_operation_fields(&snapshot))
}

#[tauri::command]
pub fn preview_codex_config(
    config_toml: String,
    fields: CodexOperationFields,
    goal_mode: Option<bool>,
) -> AppResult<String> {
    tc::codex::build_codex_config(&config_toml, &fields, goal_mode)
}

#[tauri::command]
pub fn parse_codex_fields(
    auth: Value,
    config_toml: String,
) -> AppResult<CodexOperationFields> {
    Ok(tc::codex::parse_operation_fields(&auth, &config_toml))
}

File: src/services/modules/tool_config.ts
import { request } from '../request'

/** 受支持的工具类型。 */
export type AppType = 'claude' | 'codex'

/** 渠道列表项 / 元数据。 */
export interface ChannelMeta {
  id: string
  name: string
  appType: AppType
  updatedAt: string
}

/** 抽取源配置（live）结果。 */
export interface ExtractResult {
  exists: boolean
  /** Claude=settings.json JSON；Codex={ auth, configToml, config }。 */
  snapshot: unknown
}

/** 渠道完整数据。 */
export interface ChannelData {
  id: string
  name: string
  appType: AppType
  snapshot: unknown
  updatedAt: string
}

/** Claude 操作字段。 */
export interface ClaudeOperationFields {
  baseUrl: string
  apiKey: string
  sonnetModel: string
  opusModel: string
  haikuModel: string
  defaultModel: string
}

/** Codex 操作字段。 */
export interface CodexOperationFields {
  apiKey: string
  baseUrl: string
  model: string
  reviewModel: string
}

/** Codex 渠道快照结构。 */
export interface CodexSnapshot {
  auth: Record<string, unknown>
  configToml: string
  config?: unknown
}

export interface SaveChannelRequest {
  id?: string | null
  name: string
  snapshot: unknown
}

export const toolConfigApi = {
  list: (appType: AppType) => request<ChannelMeta[]>('list_profile_channels', { appType }),
  get: (appType: AppType, id: string) => request<ChannelData>('get_profile_channel', { appType, id }),
  save: (appType: AppType, req: SaveChannelRequest) => request<ChannelMeta>('save_profile_channel', { appType, req }),
  remove: (appType: AppType, id: string) => request<void>('delete_profile_channel', { appType, id }),
  /** 读取本机真实配置 → 写 *.record.json → 返回快照。 */
  extract: (appType: AppType) => request<ExtractResult>('extract_source_record', { appType }),
  /** 备份原文件 → 原子覆写真实配置。 */
  apply: (appType: AppType, snapshot: unknown) => request<void>('apply_profile_config', { appType, snapshot }),
  previewClaude: (base: unknown, fields: ClaudeOperationFields) => request<unknown>('preview_claude_settings', { base, fields }),
  parseClaude: (snapshot: unknown) => request<ClaudeOperationFields>('parse_claude_fields', { snapshot }),
  previewCodex: (configToml: string, fields: CodexOperationFields, goalMode?: boolean) => request<string>('preview_codex_config', { configToml, fields, goalMode }),
  parseCodex: (auth: unknown, configToml: string) => request<CodexOperationFields>('parse_codex_fields', { auth, configToml }),
}
